docs/CONTRIBUTING.md

```diff
@@ -62,6 +62,35 @@ are squashed against the correct commit if necessary.
 
 ## Style Guide
 
+### Logging
+
+When possible, use log levels according to the following guidelines (in
+decreasing order of severity):
+
+- `error`: there is definitely a logic bug somewhere in the code, and someone
+  should look into it; it should not be possible to trigger an `error` log just
+  by manipulating inputs / configuration details
```

Comment:
It feels like this wording for `error` is going to be brittle and onerous for developers, because the criterion involves the holistic behaviour of the entire codebase, rather than factors local to the code at hand. 

To put it another way, deducing whether an `error` path can be hit from any possible sequence of external inputs is a tricky task, and even if the reasoning is correct at the moment when the `error` is added, there's no way to preserve that rationale as the codebase evolves.